### ISSUE TITLE
How to debug the application locally

### ISSUE THREAD

**Issue by aslary** [claudiajs/claudia#218]
After digging through the docs of Claudia and AWS, I am still not able to setup a local environment, in which I can debug the application. I am using the Claudia API Builder for the REST methods.

Claudia is great for simple deployment, but how can I debug my app locally?

**Comment by dmackinn** [comment]
@aslary have you tried using claudia-local-api ?
https://www.npmjs.com/package/claudia-local-api

You can run Claudia locally like an express app during development. Here is an npm script that I use to run mine
`"start-local": "nodemon --exec claudia-local-api --api-module api.js --port 2932 | bunyan --output short",`

**Comment by rcoundon** [comment]
> After digging through the docs of Claudia and AWS, I am still not able to setup a local environment, in which I can debug the application. I am using the Claudia API Builder for the REST methods.
> 
> Claudia is great for simple deployment, but how can I debug my app locally?
> 
> I have seen this: https://claudiajs.com/tutorials/testing-locally.html, but my goal is really to analyse my program line by line with the debugger.

Something else you could consider is Thundra.io - they have tools for allowing you to attach a debugger to a lambda function as it's actually running in AWS.

**Comment by rcoundon** [comment]
@dmackinn - have you found a way to attach a debugger to the process being run by claudia-local-api?

**Comment by dmackinn** [comment]
@rcoundon For local development using claudia-local-api it runs just like an express server and you can console.log anywhere in the app.

**Comment by rcoundon** [comment]
@dmackinn Yeah, I've used it, just haven't managed to attach a live debugger so was wondering if anyone had.
